pkg/fqdn/service/dns_handler.go

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package service
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net/netip"
+
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/fqdn"
+	"github.com/cilium/cilium/pkg/fqdn/dnsproxy"
+	"github.com/cilium/cilium/pkg/fqdn/namemanager"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+type UpdateOnDNSMsgHandler interface {
+	UpdateOnDNSMsg(lookupTime time.Time, ep *endpoint.Endpoint, qname string, responseIPs []netip.Addr, TTL int, stat *dnsproxy.ProxyRequestContext) error
+}
+
+type updateOnDNSMsgHandler struct {
```

Comment:
Would it be an option to keep the new function `UpdateOnDNSMsg` on the `DNSRequestHandler` instead of splitting it off into a separate struct. I'm fine by moving it to `pkg/fqdn/service` (or maybe even moving it to a separate package `pkg/fqdn/messagehandler` - as it's not only related to standalone) but i think these two functions should stay an on the same handler.

And then try to document the to functions and their differences

- `NotifyOnDNSMsg`
- `UpdateOnDNSMsg`

Reply:
100%. I moved it to a different package `pkg/fqdn/messagehandler`. Let me know if this looks good.